lib/file-store.js

```diff
@@ -333,172 +323,120 @@ const FileStore = function(rootDirectory) {
-      fs.readFile(destMetadataFilePath, function(err, data) {
-        if (err) {
-          return done(err);
-        }
-        done(null, buildS3ObjectFromMetaDataFile(destKey, data));
+      fs.readFile(destMetadataFilePath, (err, data) => {
+        if (err) return done(err);
+        done(null, this.buildS3ObjectFromMetaDataFile(destKey, data));
       });
     }
-  };
+  }
 
-  const deleteObject = function(bucket, key, done) {
-    const bucketPath = getBucketPath(bucket.name);
+  deleteObject(bucket, key, done) {
+    const bucketPath = this.getBucketPath(bucket.name);
     const keyPath = path.resolve(bucketPath, key);
     async.map(
-      [path.join(keyPath, METADATA_FILE), path.join(keyPath, CONTENT_FILE)],
+      [path.join(keyPath, CONTENT_FILE), path.join(keyPath, METADATA_FILE)],
       fs.unlink,
-      function(err) {
-        if (err) {
-          return done(err);
-        }
-        fs.rmdir(keyPath, function() {
-          utils.removeEmptyDirectories(fs, bucketPath, function() {
+      err => {
+        if (err) return done(err);
+        fs.rmdir(keyPath, () => {
+          utils.removeEmptyDirectories(fs, bucketPath, () => {
             return done();
           });
         });
       }
     );
-  };
-
-  const getObjectExists = function(bucket, key, done) {
-    const keyPath = path.resolve(getBucketPath(bucket.name), key);
-    fs.stat(keyPath, function(err, file) {
-      if (err || !file.isDirectory()) {
-        return done("Object not found for " + keyPath);
-      }
-      return done(null);
-    });
-  };
-
-  const concatStreams = function(passThrough, streams) {
-    const stream = streams.shift();
-    if (!stream) {
-      passThrough.end();
-      return passThrough;
-    }
-    stream.once("end", function() {
-      concatStreams(passThrough, streams);
-    });
-    stream.pipe(passThrough, { end: false });
-    return passThrough;
-  };
+  }
 
-  const combineObjectParts = function(bucket, key, uploadId, parts, req, done) {
-    const sortedParts = _.sortBy(parts, function(part) {
-      return part.number;
-    });
-    const partPaths = _.map(sortedParts, function(part) {
-      return path.resolve(
-        getBucketPath(bucket.name),
+  combineObjectParts(bucket, key, uploadId, parts, req, done) {
+    const sortedParts = sortBy(parts, part => part.number);
+    const partPaths = sortedParts.map(part => {
+      return path.join(
+        this.getBucketPath(bucket.name),
         uploadId + "_" + part.number
       );
     });
-    const partStreams = _.map(partPaths, function(partPath) {
-      return fs.createReadStream(path.join(partPath, CONTENT_FILE));
-    });
-    const combinedPartsStream = concatStreams(new PassThrough(), partStreams);
+    const partStreams = partPaths.map(partPath =>
+      fs.createReadStream(path.join(partPath, CONTENT_FILE))
+    );
+    const combinedPartsStream = this.concatStreams(
+      new PassThroughStream(),
```

Comment:
@specialkk I don't have a strong opinion about this but why assign your own name to `PassThrough`? The original name is clear enough and common to Node.js, reading this made me think that you rolled out your own version.

Reply:
I've just developed the habit of renaming them to be suffixed with `Stream` ever since destructuring imports became common. Before you'd at least be able to see
```javascript
var s = new stream.PassThrough();
```
but now it looks like
```javascript
const s = new PassThrough();
```
It's kind of a flimsy reason and I can see how it can look weird so I'll revert it.